src/powerman/client.c

```diff
@@ -669,6 +670,24 @@ static void _telemetry_printf(int client_id, const char *fmt, ...)
     }
 }
 
+/*
+ * Callback for device diagnostics
+ */
+static void _diag_printf(int client_id, const char *fmt, ...)
+{
```

Comment:
Is there a way this printf like function could be protected with `__attribute__ printf`?  Maybe it can be added to the typedef?  (I don't recall ever doing it with a pointer to a function)

Reply:
it wasn't there in the telemetry function so just didn't add it :-)   I'll add to the telemetry one in an extra commit.